### PR TITLE
dev/core#1004 Fix regression causing notices on components screen

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
Fixes unreleased regression causing notices on enable components page

Before
----------------------------------------
Notices

After
----------------------------------------
Poof

This DOES change from
advmultiselect to a normal UI - not sure if there is any reason not so, since adv multiselect is
deprecated

Screen Shot 2019-06-01 at 4 54 29 PM

Technical Details
----------------------------------------
I've cleaned up the metadata & moved relevant functions to the right places. 

----------------------------------------
@colemanw as @jaapjansma pointed out this is an unreleased regression from https://github.com/civicrm/civicrm-core/pull/14264 - I've fixed the form to use the metadata - plse review

### CODE REVIEW

File: CRM/Admin/Form/Setting/Component.php [hunk 1]
```diff
@@ -76,98 +65,20 @@ public function buildQuickForm() {
...
       }
     }
 
     return $errors;
   }
 
-  /**
-   * @return array
-   */
-  private function _getComponentSelectValues() {
```
Comment (eileenmcnaughton) [inline CRM/Admin/Form/Setting/Component.php]: moved

File: CRM/Admin/Form/Setting/Component.php [hunk 2]
```diff
@@ -76,98 +65,20 @@ public function buildQuickForm() {
...
-    CRM_Core_BAO_Navigation::resetNavigation();
-  }
-
-  /**
-   * Load case sample data.
-   *
-   * @param string $fileName
-   * @param bool $lineMode
-   */
-  public static function loadCaseSampleData($fileName, $lineMode = FALSE) {
```
Comment (eileenmcnaughton) [inline CRM/Admin/Form/Setting/Component.php]: moved

File: CRM/Admin/Form/Setting/Component.php [hunk 3]
```diff
@@ -76,98 +65,20 @@ public function buildQuickForm() {
...
-    $ret = [];
-    $this->_components = CRM_Core_Component::getComponents();
-    foreach ($this->_components as $name => $object) {
-      $ret[$name] = $object->info['translatedName'];
-    }
-
-    return $ret;
-  }
-
-  public function postProcess() {
```
Comment (eileenmcnaughton) [inline CRM/Admin/Form/Setting/Component.php]: gone - resetnav call moved

**Comment by eileenmcnaughton** [comment]
test this please

**Comment by eileenmcnaughton** [comment]
test this please

**Comment by eileenmcnaughton** [comment]
test this please

**Comment by demeritcowboy** [comment]
Hi, it gives a different warning now: `Strict warning: call_user_func_array() expects parameter 1 to be a valid callback, non-static method CRM_Core_SelectValues::getComponentSelectValues() should not be called statically in Civi\Core\Resolver->call() (line 119 of /home/jenkins/bknix-dfl/build/core-14393-2ht0e/sites/all/modules/civicrm/Civi/Core/Resolver.php).`

Also I wonder if it's possible to make the widget taller/wider? The dropdown is hidden.

**Comment by eileenmcnaughton** [comment]
thanks @demeritcowboy I made it static - I'm hoping @colemanw will comment on best UI choice here to replace the deprecated advmultiselect widget

**Comment by demeritcowboy** [comment]
Ok thanks.

**Comment by eileenmcnaughton** [comment]
@colemanw can you check this before the rc is cut

**Comment by jaapjansma** [comment]
I now get a warning. 
Screenshot from 2019-06-03 10-17-36

and this is how my components screen looks like after skipping all warnings.
Screenshot from 2019-06-03 10-18-51


ps. the index notices regarding the localisation are present on every page in civicrm and where also present before testing this PR.

**Comment by eileenmcnaughton** [comment]
@jaapjansma this would only work with a cache clear - but I'm unsure how much of the problem persists once you have done that

**Comment by jaapjansma** [comment]
Cache clear made it work for me. I forgot to clear the caches and then thought hang on lets try that out. ;-) So you can ignore my comment.

**Comment by eileenmcnaughton** [comment]
@jaapjansma do you have an opinion on the widget change - if you think it doesn't need further input from @colemanw  I'll go ahead & merge

**Comment by jaapjansma** [comment]
I think the widget change makes sense.

**Comment by eileenmcnaughton** [comment]
@jaapjansma thanks I'll merge then. The widget change was a by-product but I'd rather get the regression cleared up before it wastes any one else's time

**Comment by colemanw** [comment]
@eileenmcnaughton I've put a proposed improvement to the style which can be reused for other groups of checkboxes as well: #14419

**Pull request merged**